diff -ru Python-2.7.12-orig/configure.ac Python-2.7.12/configure.ac
--- Python-2.7.12-orig/configure.ac	2016-06-25 23:49:32.000000000 +0200
+++ Python-2.7.12/configure.ac	2017-08-05 13:24:37.538164985 +0200
@@ -1698,7 +1698,7 @@
 # structures (such as rlimit64) without declaring them. As a
 # work-around, disable LFS on such configurations
 
-use_lfs=yes
+use_lfs=no
 AC_MSG_CHECKING(Solaris LFS bug)
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #define _LARGEFILE_SOURCE 1
